examples/golang/components/http-hello-world/main.go

```diff
@@ -0,0 +1,18 @@
+package main
+
+import (
+	"net/http"
+
+	"go.wasmcloud.dev/component/net/wasihttp"
+)
+
+func init() {
+	wasihttp.HandleFunc(handleRequest)
+}
+
+func handleRequest(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte("Hello from Go!\n"))
+}
+
+//go:generate go run github.com/bytecodealliance/wasm-tools-go/cmd/wit-bindgen-go generate --world hello --out gen ./wit
+func main() {}
```

Comment:
I think we should consider moving the `//go:generate` line to the top of the file, and the `func main() { .. }` further up, maybe right after `init()`.

May even be worth including a comment explaining why it's empty, because anyone familiar with Go would expect this to be where the magic happens, so to speak.

Reply:
Fair enough, are you thinking a simple "this is because we don't call main when we call the component, we call the HTTP request handle" would be sufficient?

Reply:
Updated the help text, let me know what you think 🙂

Reply:
that seems great 👍